packages/web-components/src/popover/popover.ts

```diff
@@ -0,0 +1,129 @@
+import { attr, FASTElement, Updates } from '@microsoft/fast-element';
+import { arrow, computePosition, offset } from '@floating-ui/dom';
+
+export class Popover extends FASTElement {
+  @attr({ mode: 'boolean' })
+  public open: boolean = false;
+
+  @attr({ mode: 'boolean' })
+  public arrow: boolean = false;
+
+  @attr
+  public position: 'top' | 'bottom' | 'left' | 'right' = 'top';
+
+  public popoverContentRef: FASTElement; // @FIXME: what type?
+  public arrowRef: FASTElement;
+  public anchorRef: HTMLElement[];
+
+  public connectedCallback() {
+    super.connectedCallback();
+    this.openChanged(); // hack
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this.hidePopover();
+  }
+
+  protected openChanged() {
+    // this is from FAST tooltip: if ((this as FASTElement).$fastController.isConnected) {
+    if ((this as FASTElement).$fastController.isConnected) {
+      if (this.open) {
+        this.showPopover();
+      } else {
+        this.hidePopover();
+      }
+
+      Updates.enqueue(() => {
+        if (this.open) {
+          this.updatePosition();
+        }
+      });
+    }
+  }
+
+  private isOpen: boolean = false;
+
+  private showPopover = (): void => {
+    if (this.isOpen) {
+      return;
+    }
+
+    document.addEventListener('keydown', this.handleDocumentKeyDown);
+    document.addEventListener('click', this.handleOutsideClick, true);
+
+    this.isOpen = true;
+  };
+
+  private hidePopover = (): void => {
+    if (!this.isOpen) {
+      return;
+    }
+
+    document.removeEventListener('click', this.handleOutsideClick, true);
+    document.removeEventListener('keydown', this.handleDocumentKeyDown);
+
+    this.isOpen = false;
+  };
+
+  private handleDocumentKeyDown = (e: KeyboardEvent) => {
+    // @FIXME: should this be imported from @microsoft/fast-web-utilities?
```

Comment:
@miroslavstastny I do think we should pull in from `@microsoft/fast-web-utilities` so that we have a single source of "truth". These are pulled in via Foundation and we'd be duplicating to do it elsewhere.